### PR TITLE
kubemq: init kubemq-community at 2.2.12 and kubemqctl at 3.5.1

### PR DESCRIPTION
**Pull request by brianmcgee**
###### Description of changes

Introduces the community edition of the [KubeMQ](https://kubemq.io) server and it's associated cli. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/kubemqctl/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ lib, stdenv, buildGoModule, fetchFromGitHub, pkg-config }:
+
+buildGoModule rec {
+  pname = "kubemqctl";
+  version = "3.5.1";
+  src = fetchFromGitHub {
+    owner = "kubemq-io";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256:0daqvd1y6b87xvnpdl2k0sa91zdmp48r0pgp6dvnb2l44ml8a4z0";
```
Comment (zimbatm) [inline pkgs/applications/networking/cluster/kubemqctl/default.nix:10]:
```suggestion
    sha256 = "0daqvd1y6b87xvnpdl2k0sa91zdmp48r0pgp6dvnb2l44ml8a4z0";
```

File: pkgs/applications/networking/cluster/kubemqctl/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{ lib, stdenv, buildGoModule, fetchFromGitHub, pkg-config }:
```
Comment (zimbatm) [inline pkgs/applications/networking/cluster/kubemqctl/default.nix:1]:
```suggestion
{ lib, stdenv, buildGoModule, fetchFromGitHub }:
```

File: pkgs/servers/kubemq-community/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{ lib, stdenv, buildGoModule, fetchFromGitHub, pkg-config }:
```
Comment (zimbatm) [inline pkgs/servers/kubemq-community/default.nix:1]:
```suggestion
{ lib, stdenv, buildGoModule, fetchFromGitHub }:
```

File: pkgs/servers/kubemq-community/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    rev = "v${version}";
+    sha256 = "sha256:06n3avcqknqzf9y03xqcsg36pwcha29j2psp9xsnir7hrx66zww8";
+  };
+
+  CGO_ENABLED=0;
+  LDFLAGS="-w -s -X main.version=${version}";
+
+  doCheck=false;  # grpc tests are flaky
+
+  vendorSha256 = "sha256:1sh0dzz8z065964k2gzkzw9p3db3rcf6mv901zym0wqm4p71045w";
```
Comment (zimbatm) [inline pkgs/servers/kubemq-community/default.nix:18]:
```suggestion
  vendorSha256 = "1sh0dzz8z065964k2gzkzw9p3db3rcf6mv901zym0wqm4p71045w";
```

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -26985,6 +26987,8 @@ with pkgs;
   kubernetes = callPackage ../applications/networking/cluster/kubernetes { };
   kubectl = callPackage ../applications/networking/cluster/kubernetes/kubectl.nix { };
 
+  kubemqctl = callPackage ../applications/networking/cluster/kubemqctl/default.nix { };
```
Comment (zimbatm) [inline pkgs/top-level/all-packages.nix:26990]:
```suggestion
  kubemqctl = callPackage ../applications/networking/cluster/kubemqctl { };
```

**Comment by zimbatm** [comment]
Result of `nixpkgs-review pr 163657` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>kubemq-community</li>
    <li>kubemqctl</li>
  </ul>
</details>

File: pkgs/applications/networking/cluster/kubemqctl/default.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  pname = "kubemqctl";
+  version = "3.5.1";
+  src = fetchFromGitHub {
+    owner = "kubemq-io";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256:0daqvd1y6b87xvnpdl2k0sa91zdmp48r0pgp6dvnb2l44ml8a4z0";
+  };
+
+  LDFLAGS="-X main.version=${version}";
```
Comment (Mic92) [inline pkgs/applications/networking/cluster/kubemqctl/default.nix:13]:
```suggestion
  ldflags = [ "-X main.version=${version}" ];
```

File: pkgs/applications/networking/cluster/kubemqctl/default.nix [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+  doCheck=false; # TODO tests are failing
```
Comment (Mic92) [inline pkgs/applications/networking/cluster/kubemqctl/default.nix:15]:
```suggestion
  doCheck = false; # TODO tests are failing
```

File: maintainers/maintainer-list.nix [hunk 8]
```diff
@@ -14112,4 +14112,10 @@
     github = "nigelgbanks";
     githubId = 487373;
   };
+  brianmcgee = {
```
Comment (zowoq) [inline maintainers/maintainer-list.nix:14115]: This should be a separate commit: `maintainers: add brianmcgee`

https://github.com/NixOS/nixpkgs/blob/a15fbdb88f38800c8e554772a12de8031815802a/maintainers/maintainer-list.nix#L49

Please sort the new entry alphabetically.

File: nixos/doc/manual/from_md/release-notes/rl-2205.section.xml [hunk 9]
```diff
@@ -105,6 +105,22 @@
           a kernel module for mounting the Apple File System (APFS).
         </para>
       </listitem>
+      <listitem>
```
Comment (zowoq) [inline nixos/doc/manual/from_md/release-notes/rl-2205.section.xml:108]: Release notes aren't needed for new packages.

File: pkgs/applications/networking/cluster/kubemqctl/default.nix [hunk 10]
```diff
@@ -0,0 +1,25 @@
+{ lib, stdenv, buildGoModule, fetchFromGitHub }:
```
Comment (zowoq) [inline pkgs/applications/networking/cluster/kubemqctl/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:
```

File: pkgs/servers/kubemq-community/default.nix [hunk 11]
```diff
@@ -0,0 +1,26 @@
+{ lib, stdenv, buildGoModule, fetchFromGitHub }:
```
Comment (zowoq) [inline pkgs/servers/kubemq-community/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:
```

File: pkgs/servers/kubemq-community/default.nix [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+    owner = "kubemq-io";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "06n3avcqknqzf9y03xqcsg36pwcha29j2psp9xsnir7hrx66zww8";
+  };
+
+  CGO_ENABLED=0;
+  LDFLAGS="-w -s -X main.version=${version}";
+
+  doCheck=false;  # grpc tests are flaky
```
Comment (zowoq) [inline pkgs/servers/kubemq-community/default.nix:16]:
```suggestion
  CGO_ENABLED = 0;
  ldflags = [ "-w" "-s" "-X main.version=${version}" ];

  doCheck = false;  # grpc tests are flaky
```

**Review by zowoq — COMMENTED** [review]
Please squash this down to three commits, one for the new maintainer and one commit per package.

**Comment by zowoq** [comment]
Could you fix the commit message for the two packages please, both are missing an `e`. Our CI will build them automatically if the commit messages match.

**Comment by brianmcgee** [comment]
> Could you fix the commit message for the two packages please, both are missing an `e`. Our CI will build them automatically if the commit messages match.

Well that was stupid :( Thanks for all the feedback
